[Search] 'All' checked upon no filters. Fixes #1117

vankessel opened the pull request:
# Checklist
* Changes address original issue?
 * Yes. 'All' gets checked when all filters are unchecked.
* Unit tests included and/or updated with changes?
 * Unit test for SearchMenuController has been updated to test for these changes, and succeeds when run. 
* Command line build passes?
 * Yes.
* Changes have been smoke-tested?
 * Yes.

VWoeltjen commented:
Thanks for the contribution - changes look good! 

Before we can integrate open source contributions, we need a signed Contributor License Agreement. Email us at ARC-DL-OpenMCT@mail.nasa.gov and we will send this to you.

akhenry commented:
@vankessel Nice work, this is a clean, succinct fix, with the added bonus of no net code increase :) I just have one comment on a test which I'll put inline.

Once that's addressed and we have your signed CLA I'll merge this in.

akhenry commented on platform/search/test/controllers/SearchMenuControllerSpec.js:
```diff
@@ -83,7 +83,7 @@ define(
                 mockScope.ngModel.checkAll = false;
 
                 controller.checkAll();
-                expect(mockScope.ngModel.filtersString).toEqual('NONE');
```
The description of this test is a little inconsistent with its implementation now. I wonder if we should repurpose it to test that the "All" option cannot be unset when nothing is selected.

I'll leave it up to you how best to implement this. It could be as simple as taking the existing test and just adding some lines to verify that `ngModel.checkAll` is still true after each invocation of `controller.checkAll()`. You might also want to add a check on the precondition that nothing is selected.

vankessel commented:
@akhenry Thank you! I've addressed the comment and signed the CLA. I added a few lines to check for checkAll being true after each invocation, but I didn't add the check that nothing is selected as the test below the one in question ensures that the other options are false after invocations of checkAll()

akhenry commented:
👍  Looks good to me. Thanks again!

### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
